src/app/index.cpp

```diff
@@ -160,6 +164,64 @@ namespace app::index
 
   void KVIndexer::compact(int64_t at)
   {
-    // TODO(#63): compact entries before `at`
+    std::shared_lock lock(mutex);
+
+    // get the first revision still in the index
+    auto start_revision = revisions_to_key.begin()->first;
+    CCF_APP_DEBUG("Compacting index from {} to {}", start_revision, at);
+
+    std::set<K> keys_compacted;
```

Comment:
Why not deleting compacted values from `keys_to_values` in the same loop at the first one, rather than doing it in a separate loop below?

Reply:
I think in future it would be nicer to be able to remove potentially multiple values for a key in one go to avoid the copying of vector elements. This way makes that change easier.

Reply:
Will add a comment